sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/credential/CacheableTokenCredential.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.identity.credential;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.spring.cloud.service.implementation.identity.Cache;
+import com.azure.spring.cloud.service.implementation.identity.credential.adapter.CacheKeyDescriptor;
+import reactor.core.publisher.Mono;
+
+/**
+ * TokenCredential that can cache AccessToken.
+ */
+public class CacheableTokenCredential implements TokenCredential {
```

Comment:
I'm not happy with the caching design here, I think you're making too many places couple with the cache, ideally it should only one please aware of the caching exist
let's discuss this offline sometime and see we can improve it

Reply:
Yes, I know there are a lot of caches here,  we can discuss this sometime.